### DISCUSSION TITLE
delete more than records

### DISCUSSION THREAD

**Discussion by samaremad** [serenity-is/Serenity#5463] [Help]
i want the function of delete to cheack if used the id of that rcords in inother table then delte it also from that recored 

 public DeleteResponse Delete(IUnitOfWork uow, DeleteRequest request)
        {
            var n = request.EntityId;

            int convertKey = Convert.ToInt32(n);
            List<voteRecord> v;
            if(db.voteRecords.FirstOrDefault(x => x.id_voteSub.Equals(convertKey))!= null)
            {
                v = db.voteRecords.Where(x => x.id_voteSub.Equals(convertKey)).ToList();
                db.voteRecords.Remove(v);
            }
}

how can i do that

**Comment by minhhungit** [comment]
Try this way, write code in **Repository.cs ==> MyDeleteHandler** 

```csharp
private class MyDeleteHandler : DeleteRequestHandler<MyRow>
{
    protected override void OnBeforeDelete()
    {
        base.OnBeforeDelete();

        /* list products by CategoryID */
        var products = this.UnitOfWork.Connection.List<Entities.ProductRow>(new Criteria(Entities.ProductRow.Fields.CategoryID) == (int)Row.CategoryID);
        foreach (var prodRow in products)
        {
            this.UnitOfWork.Connection.DeleteById<Entities.ProductRow>(prodRow.ProductID);
        }
    }
}
```
